### DISCUSSION TITLE
Add new-line parameter to COPY TO CSV like already exist in COPY TO

### DISCUSSION THREAD

**Discussion by bladko-tom** [duckdb/duckdb#7479] [Ideas]
Idea to add "new_line" parameter to setup character for new line like '\r' or \r\n' to COPY TO CSV as it already exist in read_csv() function.

https://duckdb.org/docs/data/csv/overview.html#read_csv_auto-function

**Comment by lnkuiper** [comment]
Do you mean this?
```sql
copy (select range from range(10)) to 'my.csv' (new_line '\n');
```

It [does not seem to be documented](https://duckdb.org/docs/sql/statements/copy.html).

**Reply by bladko-tom** [reply]
but it seams it does not catch parameter actually, i tried several options it always use LF or \n

**Reply by pdet** [reply]
Yes, you are absolutely correct. This parameter is only used in the read_csv. I'm happy to review a PR, if you would like to give it a go :-)

**Reply by Mytherin** [reply]
It's actually used in the CSV writer as well as per #7478 so this should already work in the latest master:

```sql
COPY (FROM range(5)) TO 'out.csv' (NEW_LINE '-');
```

```bash
$ > cat out.csv
0-1-2-3-4-
```

**Reply by bladko-tom** [reply]
@Mytherin  thanks but how to make this works then 

`
 COPY (FROM range(5)) TO 'out.csv' (NEW_LINE '\r\n');
 `
 This creates a file with literal \r\n not as special chars.

**Reply by theholy7** [reply]
Seems like there is a caveat when you pass the terminator as plain SQL text or as an escaped string `E'\n'` for example.

**Comment by bladko-tom** [comment]
Still cannot find this in docs

**Reply by Tishj** [reply]
Feel free to make an issue on the docs if this is just a docs related issue 👍

**Reply by theholy7** [reply]
I documented this today @Tishj - I think we can close this.

**Comment by theholy7** [comment]
Has this functionality been confirmed as real and ready to use?

https://github.com/duckdb/duckdb/discussions/22995
